src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -22,12 +22,23 @@ public static int IndexOf(ref char searchSpace, int searchSpaceLength, ref char
             if (valueLength == 0)
                 return 0;  // A zero-length sequence is always treated as "found" at the start of the search space.
 
+            int index = 0;
             char valueHead = value;
-            ref char valueTail = ref Unsafe.Add(ref value, 1);
             int valueTailLength = valueLength - 1;
+
+            // Try "Algorithm 1: Generic SIMD" as described in http://0x80.pl/articles/simd-strfind.html#algorithm-1-generic-simd
+            // for values longer than 2 chars
+            if (valueLength > 1)
```

Comment:
Should this be `valueTailLength > 1`, or the comment "for values longer than 1 char"?